### ISSUE TITLE
can ansible operator watch a specific k8s resource

### ISSUE THREAD

**Issue by aravinder111** [operator-framework/operator-sdk#3094]
## Type of question

**Are you asking about community best practices, how to implement a specific feature, or about general context and help around the operator-sdk?**
Help with operator-sdk's Ansible operator pattern

## Question

**What did you do?**
I would like to watch a specific k8s resource, to be specific I want to watch a configmap which is not created by my role. Whenever there is a change to the configmap, I want to reconcile my role.

**What did you expect to see?**
An optional field `name` to be specified in watches.yaml

**What did you see instead? Under which circumstances?**
I dont see an option to watch a specific resource by its name in documentation https://sdk.operatorframework.io/docs/ansible/reference/watches/

**Environment**
* operator-sdk version:

```
operator-sdk version: "v0.17.0", commit: "2fd7019f856cdb6f6618e2c3c80d15c3c79d1b6c", kubernetes version: "v1.17.2", go version: "go1.14.2 darwin/amd64"
```

* Kubernetes version information:

 ```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.9+IKS", GitCommit:"044064595e3a0af0d71be49184f82bdfc69924ac", GitTreeState:"clean", BuildDate:"2020-04-17T15:25:39Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by aravinder111** [comment]
Slack conversation regarding the same is [here](https://kubernetes.slack.com/archives/CAW0GV7A5/p1590165384414000)

**Comment by camilamacedo86** [comment]
With Ansible operator is possible to watch the resources which are dependent. See; https://sdk.operatorframework.io/docs/ansible/reference/dependent-watches/

But, confirmed with @fabianvf that currently has not a way to scope the operator to a single resource (other than checking and ending early from Ansible)

So, it will be flagged as RFE and will be discussed in the next triage meeting.

**Comment by camilamacedo86** [comment]
/remove-lifecycle rotten

c/c @asmacdo @fabianvf

**Comment by fabianvf** [comment]
@estroz is this supported from golang operators (I assume it is?). If so, we do probably want to add this functionality to keep parity

**Comment by estroz** [comment]
Yes if a predicate is used that filters by name. Perhaps allowing multiple `names` per GVK?

Side note: requests like this are really good reasons to use a Go-Ansible hybrid operator; this particularly request is easy to implement and the watches API change is small, so conversion to a hybrid operator isn’t necessary.

**Comment by absynth76** [comment]
small remark: I think it's implemented using predicates on golang operator, but it's at reconciliation time that the decision is made to pick up or not the resource. It's also possible with ansible, you can watch all configmaps and execute your action only if the configmap respect your requirements (name/annotation/label, you only have to query for the configmap and filter according to your needs) I don't think this is optimal because you'll actually watch all configmaps, but it does work.

**Comment by niuxiaojuan-github** [comment]
We also need this feature, can this be implemented soon?

**Comment by laurieaw63** [comment]
> Side note: requests like this are really good reasons to use a Go-Ansible hybrid operator; this particularly request is easy to implement and the watches API change is small, so conversion to a hybrid operator isn’t necessary.

Is there an example somewhere that demonstrates a hybrid?
